atomic_reactor/plugin.py

```diff
@@ -258,10 +274,21 @@ def run(self, keep_going=False):
                 logger.exception("failed to save plugin duration")
 
             self.plugins_results[plugin_class.key] = plugin_response
+
+            if plugin_successful and buildstep_phase:
+                logger.debug('stopping further execution of plugins '
+                             'after first successful plugin')
+                break
+
         if len(failed_msgs) == 1:
             raise PluginFailedException(failed_msgs[0])
         elif len(failed_msgs) > 1:
             raise PluginFailedException("Multiple plugins raised an exception: " + str(failed_msgs))
+
+        if (plugins_skipped == len(self.plugins_conf)) and buildstep_phase:
```

Comment:
In that case, `plugin_successful` wouldn't be set to `True`, right?

If a build phase plugin fails and is allowed to fail (valid scenario??), then `plugins_skipped` is not incremented. In which case, this method wouldn't raise any exceptions. To me that seems incorrect.